nodes/src/nodes/tool_exa_search/exa_driver.py

```
@@ -0,0 +1,343 @@
+
+    def _tool_validate(self, *, tool_name: str, input_obj: Any) -> None:  # noqa: ANN401
+        bare = self._bare_name(tool_name)
+        if bare != 'exa_search':
+            raise ValueError(f'Unknown tool {tool_name!r} (expected exa_search)')
+
+        if not isinstance(input_obj, dict):
+            raise ValueError('Tool input must be a JSON object')
+
+        query = input_obj.get('query')
+        if not query or not isinstance(query, str) or not query.strip():
+            raise ValueError('query is required and must be a non-empty string')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)